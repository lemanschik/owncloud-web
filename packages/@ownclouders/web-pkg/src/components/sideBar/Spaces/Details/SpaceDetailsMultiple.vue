<template>
  <div id="oc-spaces-details-multiple-sidebar">
    <div class="spaces-preview oc-mb">
      <div class="spaces-preview-body">
        <oc-icon class="preview-icon" size="xxlarge" variation="passive" name="layout-grid" />
        <p class="preview-text" v-text="selectedSpacesString" />
      </div>
    </div>
    <div>
      <table class="details-table" :aria-label="detailsTableLabel">
        <tr>
          <th scope="col" class="oc-pr-s oc-font-semibold" v-text="$gettext('Total quota:')" />
          <td v-text="totalSelectedSpaceQuotaTotal" />
        </tr>
        <tr>
          <th scope="col" class="oc-pr-s oc-font-semibold" v-text="$gettext('Remaining quota:')" />
          <td v-text="totalSelectedSpaceQuotaRemaining" />
        </tr>
        <tr>
          <th scope="col" class="oc-pr-s oc-font-semibold" v-text="$gettext('Used quota:')" />
          <td v-text="totalSelectedSpaceQuotaUsed" />
        </tr>
        <tr>
          <th scope="col" class="oc-pr-s oc-font-semibold" v-text="$gettext('Enabled:')" />
          <td v-text="totalEnabledSpaces" />
        </tr>
        <tr>
          <th scope="col" class="oc-pr-s oc-font-semibold" v-text="$gettext('Disabled:')" />
          <td v-text="totalDisabledSpaces" />
        </tr>
      </table>
    </div>
  </div>
</template>
<script lang="ts">
import { formatFileSize } from 'web-pkg/src/helpers'
import { computed, defineComponent, PropType } from 'vue'
import { SpaceResource } from 'web-client/src'
import { useGettext } from 'vue3-gettext'

export default defineComponent({
  name: 'SpaceDetailsMultiple',
  props: {
    selectedSpaces: {
      type: Array as PropType<Array<SpaceResource>>,
      required: true
    }
  },
  setup(props) {
    const {
      $gettext,
      $ngettext,
      interpolate: $gettextInterpolate,
      current: currentLanguage
    } = useGettext()
    const totalSelectedSpaceQuotaTotal = computed(() => {
      let total = 0
      props.selectedSpaces.forEach((space) => {
        total += space.spaceQuota.total
      })
      return formatFileSize(total, currentLanguage)
    })
    const totalSelectedSpaceQuotaRemaining = computed(() => {
      let remaining = 0
      props.selectedSpaces.forEach((space) => {
        if (space.disabled) {
          return
        }
        remaining += space.spaceQuota.remaining
      })
      return formatFileSize(remaining, currentLanguage)
    })
    const totalSelectedSpaceQuotaUsed = computed(() => {
      let used = 0
      props.selectedSpaces.forEach((space) => {
        if (space.disabled) {
          return
        }
        used += space.spaceQuota.used
      })
      return formatFileSize(used, currentLanguage)
    })
    const totalEnabledSpaces = computed(() => {
      return props.selectedSpaces.filter((s) => !s.disabled).length
    })
    const totalDisabledSpaces = computed(() => {
      return props.selectedSpaces.filter((s) => s.disabled).length
    })
    const detailsTableLabel = computed(() => {
      return $gettext('Overview of the information about the selected spaces')
    })
    const selectedSpacesString = computed(() => {
      return $gettextInterpolate(
        $ngettext(
          '%{ itemCount } space selected',
          '%{ itemCount } spaces selected',
          props.selectedSpaces.length
        ),
        {
          itemCount: props.selectedSpaces.length
        }
      )
    })
    return {
      detailsTableLabel,
      selectedSpacesString,
      totalSelectedSpaceQuotaTotal,
      totalSelectedSpaceQuotaRemaining,
      totalSelectedSpaceQuotaUsed,
      totalEnabledSpaces,
      totalDisabledSpaces
    }
  }
})
</script>
<style lang="scss">
.spaces-preview {
  position: relative;
  background-color: var(--oc-color-background-muted);
  border: 10px solid var(--oc-color-background-muted);
  height: 230px;
  text-align: center;
  color: var(--oc-color-swatch-passive-muted);

  &-body {
    margin: 0;
    position: absolute;
    top: 50%;
    left: 50%;
    -ms-transform: translate(-50%, -50%);
    transform: translate(-50%, -50%);

    .preview-icon {
      display: inline-block;
    }
    .preview-text {
      display: block;
    }
  }
}
.details-table {
  text-align: left;

  tr {
    height: 1.5rem;
  }
}
</style>
