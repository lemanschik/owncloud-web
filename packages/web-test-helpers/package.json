{
  "name": "web-test-helpers",
  "private": true,
  "main": "src/index.ts",
  "peerDependencies": {
    "@casl/ability": "^6.3.3",
    "@casl/vue": "^2.2.1",
    "axios": "0.27.2",
    "design-system": "@ownclouders/design-system",
    "vue3-gettext": "^2.3.3",
    "vue-router": "4.1.6",
    "vuex": "4.1.0"
  }
}
