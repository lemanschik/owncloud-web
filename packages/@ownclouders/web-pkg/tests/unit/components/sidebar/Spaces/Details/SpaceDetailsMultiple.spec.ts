import SpaceDetailsMultiple from 'web-pkg/src/components/sideBar/Spaces/Details/SpaceDetailsMultiple.vue'
import { defaultPlugins, shallowMount } from 'web-test-helpers'

const spaceMock = {
  type: 'space',
  name: ' space',
  id: '1',
  mdate: 'Wed, 21 Oct 2015 07:28:00 GMT',
  spaceQuota: {
    used: 100,
    total: 1000,
    remaining: 900
  }
}

describe('Multiple Details SideBar Panel', () => {
  it('displays the details side panel', () => {
    const { wrapper } = createWrapper(spaceMock)
    expect(wrapper.html()).toMatchSnapshot()
  })
})

function createWrapper(spaceResource) {
  return {
    wrapper: shallowMount(SpaceDetailsMultiple, {
      global: {
        plugins: [...defaultPlugins()]
      },
      props: {
        selectedSpaces: [spaceResource]
      }
    })
  }
}
