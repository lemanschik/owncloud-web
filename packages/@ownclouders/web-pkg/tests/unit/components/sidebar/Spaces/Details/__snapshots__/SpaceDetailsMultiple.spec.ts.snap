// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Multiple Details SideBar Panel displays the details side panel 1`] = `
<div id="oc-spaces-details-multiple-sidebar">
  <div class="spaces-preview oc-mb">
    <div class="spaces-preview-body">
      <oc-icon-stub accessiblelabel="" class="preview-icon" color="" filltype="fill" name="layout-grid" size="xxlarge" type="span" variation="passive"></oc-icon-stub>
      <p class="preview-text">1 space selected</p>
    </div>
  </div>
  <div>
    <table aria-label="Overview of the information about the selected spaces" class="details-table">
      <tbody>
        <tr>
          <th class="oc-pr-s oc-font-semibold" scope="col">Total quota:</th>
          <td>1 kB</td>
        </tr>
        <tr>
          <th class="oc-pr-s oc-font-semibold" scope="col">Remaining quota:</th>
          <td>900 B</td>
        </tr>
        <tr>
          <th class="oc-pr-s oc-font-semibold" scope="col">Used quota:</th>
          <td>100 B</td>
        </tr>
        <tr>
          <th class="oc-pr-s oc-font-semibold" scope="col">Enabled:</th>
          <td>1</td>
        </tr>
        <tr>
          <th class="oc-pr-s oc-font-semibold" scope="col">Disabled:</th>
          <td>0</td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
